### PR TITLE
deprecation: use to_unitary rather than as_unitary.

### PR DESCRIPTION
**Pull request by krneta**
*Issue #, if available:*

*Description of changes:*
This is just a reworking of https://github.com/aws/amazon-braket-sdk-python/pull/317 as a way to deprecate the method.

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-braket-sdk-python/blob/main/README.md) and [API docs](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/unit_tests/braket/circuits/test_circuit.py [hunk 1]
```diff
@@ -1012,6 +1012,376 @@ def test_as_unitary_one_gate_returns_expected_unitary(circuit, expected_unitary)
...
+    assert np.allclose(
+        circuit.to_unitary(),
+        np.kron(gates.CPhaseShift(0.15).to_matrix(), gates.Si().to_matrix()),
+    )
+
+
+@pytest.mark.parametrize(
+    "circuit,expected_unitary",
+    [
+        (Circuit().h(0), gates.H().to_matrix()),
```
Comment (mbeach-aws) [inline test/unit_tests/braket/circuits/test_circuit.py:1101]: could we also add a test for non-contiguous qubits? 
Even just adding `Circuit().h(1), Circuit().h(2)` as test cases could work

Comment (krneta) [reply test/unit_tests/braket/circuits/test_circuit.py:1101]: added.

File: src/braket/circuits/circuit.py [hunk 2]
```diff
@@ -1089,13 +1096,52 @@ def as_unitary(self) -> np.ndarray:
                    [ 0.70710678+0.j, -0.70710678+0.j,  0.        +0.j,
                      0.        +0.j]])
         """
+        warnings.warn(
```
Comment (kshitijc) [inline src/braket/circuits/circuit.py:1099]: Could we add `category=DeprecationWarning` to warnings.warn ?

File: src/braket/circuits/circuit.py [hunk 3]
```diff
@@ -1089,13 +1096,52 @@ def as_unitary(self) -> np.ndarray:
...
+            circuit as a unitary. For an empty circuit, an empty numpy array is returned
+            (`array([], dtype=complex)`)
+
+        Raises:
+            TypeError: If circuit is not composed only of `Gate` instances,
+                i.e. a circuit with `Noise` operators will raise this error.
+
+        Examples:
+            >>> circ = Circuit().h(0).cnot(0, 1)
+            >>> circ.as_unitary()
```
Comment (kshitijc) [inline src/braket/circuits/circuit.py:1130]: to_unitary

File: src/braket/circuits/circuit.py [hunk 4]
```diff
@@ -1063,7 +1064,7 @@ def to_ir(self) -> Program:
         )
 
     def as_unitary(self) -> np.ndarray:
-        """
+        r"""
         Returns the unitary matrix representation of the entire circuit.
```
Comment (kshitijc) [inline src/braket/circuits/circuit.py:1068]: Could we update to say `in little endian` format here ?

File: src/braket/circuits/circuit.py (hunk 2 above)
Comment (krneta) [reply src/braket/circuits/circuit.py:1099]: Good call!

File: src/braket/circuits/circuit.py (hunk 3 above)
Comment (krneta) [reply src/braket/circuits/circuit.py:1130]: Good catch!

File: src/braket/circuits/circuit.py (hunk 4 above)
Comment (krneta) [reply src/braket/circuits/circuit.py:1068]: Will do!

**Review by mbeach-aws — APPROVED** [review]

**Review by kshitijc — APPROVED** [review]

**Pull request merged**